cmake/platform.cmake

```diff
@@ -143,13 +155,20 @@ elseif(UNIX OR MINGW)
         append(CMAKE_CCXX_NOWARN_FLAGS "-diag-disable:15335")
         # disable: foo has been targeted for automatic cpu dispatch
         append(CMAKE_CCXX_NOWARN_FLAGS "-diag-disable:15009")
+        # disable optimizations in debug mode
+        append(CMAKE_CXX_FLAGS_DEBUG "-O0")
     endif()
+    # disable assertions
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DNDEBUG")
```

Comment:
CMake passes `NDEBUG` in the release configuration by default. Are there cases when we have to handle this manually?

Reply:
At some point in the past it was needed for some reason, but not anymore. So this can be removed.

Reply:
Closing the discussion as the changes [are not relevant anymore](https://github.com/intel/mkl-dnn/pull/572#issuecomment-544695471)